packages/@uppy/tus/types/index.d.ts

```diff
@@ -1,22 +1,26 @@
 import type { PluginOptions, BasePlugin } from '@uppy/core'
 import type { UploadOptions } from 'tus-js-client'
 
-  type TusUploadOptions = Pick<UploadOptions, Exclude<keyof UploadOptions,
-    | 'fingerprint'
-    | 'metadata'
-    | 'onProgress'
-    | 'onChunkComplete'
-    | 'onSuccess'
-    | 'onError'
-    | 'uploadUrl'
-    | 'uploadSize'
-  >>
+type TusUploadOptions = Pick<UploadOptions, Exclude<keyof UploadOptions,
+  | 'fingerprint'
+  | 'metadata'
+  | 'onProgress'
+  | 'onChunkComplete'
+  | 'onShouldRetry'
+  | 'onSuccess'
+  | 'onError'
+  | 'uploadUrl'
+  | 'uploadSize'
+>>
+
+type Next = (err: Error) => boolean
```

Comment:
We probably want Typescript not to freak out if the user tries to pass all tus params back to Uppy.
```suggestion
type Next = (err: Error | undefined, retryAttempt?: number, options?: TusOptions) => boolean
```